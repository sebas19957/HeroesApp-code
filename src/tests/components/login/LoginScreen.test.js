import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('pruebas en <LoginScreen/>', () => {

    const userContext = {
        dispatch: jest.fn(),
        user : {
            logged: false
        }
    }

    const histoty = {
        replace: jest.fn(),
    }
    
    const wrapper = mount(
        <AuthContext.Provider value={userContext}>
            <LoginScreen history={histoty}/>
        </AuthContext.Provider>
    );

    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.container').exists()).toBe(true);
    });

    test('dede de realizar el dispatch y la navegacion', () => {
        
        wrapper.find('button').prop('onClick')();

        expect(userContext.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'Sebastian'
            }
        });

        expect(histoty.replace).toHaveBeenCalled();
    });
    
    

});
