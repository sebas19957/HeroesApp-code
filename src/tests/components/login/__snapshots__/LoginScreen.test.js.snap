// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <LoginScreen/> debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
